TTS/trainer.py

```diff
@@ -591,7 +591,17 @@ def _optimize(
 
         step_start_time = time.time()
         # zero-out optimizer
-        optimizer.zero_grad()
+        if hasattr(self.model, "capacitron_vae"):
```

Comment:
We need to keep all the model-specific settings out of the trainer as trained soon is gonna leave 🐸TTS as a separate repo.

Reply:
So I guess all these lines can happen in `get_optimizer` function in the model.

Reply:
I'm not sure if I can do that in the `get_optimizer` method, check the lines below this. Do you have any ideas how we could extract this logic?

Reply:
I don't see why not. All attributes you need are already in the model if I am not missing something.